crates/virtio-queue/src/queue.rs

```diff
@@ -611,8 +611,18 @@ impl QueueOwnedT for Queue {
         M: Deref,
         M::Target: GuestMemory,
     {
-        self.avail_idx(mem.deref(), Ordering::Acquire)
-            .map(move |idx| AvailIter::new(mem, idx, self))
+        let idx = self.avail_idx(mem.deref(), Ordering::Acquire)?;
+
+        // The number of descriptor chain heads to process should always
```

Comment:
I don't think this comment or that in the commit message is quite right. The VMM isn't the part that can cause the DoS it's the guest.

Reply:
Thank you for letting me know the message was misleading. That was not what I intended to say. 

Both the comments and commit message are updated. PTAL.